packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -333,9 +335,66 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
         return editor;
     }
 
+    /**
+     * This method finds the Link mark in the pasted content and update its Text node.
+     * If there is no text node, pass the node's fragment recursively and updates only node containing Link mark.
+     * If the Text node does not contains Link mark, push the same node to `updatedNodes`.
+     *
+     * @param fragment Fragment containing the pasted content. [Fragment](https://prosemirror.net/docs/ref/#model.Fragment)
+     * @returns modified fragment from the `updatedNode` after updating the valid link text with its href value.
+     */
+    private readonly updateLinkNodes = (fragment: Fragment): Fragment => {
+        const updatedNodes: Node[] = [];
+
+        fragment.forEach(node => {
+            if (node.isText && node.marks.length > 0) {
+                let textNode = node;
+                node.marks.forEach(mark => {
```

Comment:
Do you really expect to have multiple `link` marks for a specific node? Is that possible? If so what tests covers those cases?

If not could this be written to show the assumption that a node only has one link mark, ie an algorithm like:

```
const mark = node.marks.find(mark => mark.type.name === 'link' && mark.attrs);
if (mark) {
  if (this.validAbsoluteLinkRegex.test(mark.attrs.href as string)) {
      updatedNodes.push()
  } else {
      updatedNodes.push()
  }
} else {
    updatedNodes.push(node)
}
```

Unless I'm misunderstanding it doesn't seem like iterating across all the marks is valuable.

Reply:
Yes, you're correct. It is not possible to have multiple `link` marks in a single `text` node and it is unnecessary to loop through the marks just to identify and update the `link` mark. So I have made changes to the logic as you suggested and it works as expected 👍